nilearn/input_data/nifti_masker.py

```diff
@@ -14,7 +14,10 @@
 
 
 class NiftiMasker(BaseMasker, CacheMixin):
-    """Nifti data loader with preprocessing
+    """Helper for masking of Niimg-like objects. NiftiMasker is useful when
+    preprocessing (detrending, standardization, resampling, etc.) of in-mask
+    voxels is necessary. Use case: working with time series of resting-state
+    or task maps.
```

Comment:
This is helpful!  Adding `intersect_masks` to related functions would be appreciated as well.

I also suggest adding to the docstring of `fit`, to reminds users that only `mask_args` and `mask_strategy` are used for fitting.

Lastly, I would find it helpful to specify (here or in `transform`) that:
* Parameters (detrend, standardize) are computed from the images passed to `transform` (i.e. not computed and stored from `fit`)
* The inverse of these parameters is not applied when `inverse_transform` is called.

Reason for those last two suggestions: as someone coming from computer vision, I believe the setup here differs in the two ways above from how we (at least in our lab) preprocess our datasets.  So my expectations may be different than someone who comes from a neuroimaging background.